src/ftpdctl.c

```diff
@@ -355,7 +355,10 @@ int main(int argc, char *argv[]) {
   (void) close(sockfd);
   destroy_pool(ctl_pool);
 
-  return 0;
+  /* Controls action status values are zero or negative, thus we need to
+   * convert them to appropriate Unix exit status values.
+   */
+  return (status * -1);
```

Comment:
Here is where we actually _use_ the richer Controls API status values for the exit status.